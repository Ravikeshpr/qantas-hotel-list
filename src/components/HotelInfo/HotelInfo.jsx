import React from "react";
import {
    HotelInfoContainer,
    HotelNameAndAddressContainer,
    NameAndRatingContainer,
    StyledAdress,
    StyledCancellationPolicy,
    StyledName,
    StyledOffer,
} from "./HotelInfo.style";
import HotelRating from "../HotelRating/HotelRating";
import { snakeCaseToTitleCase } from "../../utils/stringUtils";

function HotelInfo({
    address = [],
    name,
    rating,
    offerName,
    cancellationPolicy,
}) {
    return (
        <HotelInfoContainer>
            <NameAndRatingContainer>
                <HotelNameAndAddressContainer>
                    <StyledName>{name}</StyledName>
                    <StyledAdress>{address?.join(", ")}</StyledAdress>
                </HotelNameAndAddressContainer>

                <HotelRating
                    value={rating?.ratingValue}
                    type={rating?.ratingType}
                />
            </NameAndRatingContainer>

            <StyledOffer>{offerName}</StyledOffer>
            <StyledCancellationPolicy>
                {cancellationPolicy
                    ? snakeCaseToTitleCase(cancellationPolicy)
                    : ""}
            </StyledCancellationPolicy>
        </HotelInfoContainer>
    );
}

export default HotelInfo;
