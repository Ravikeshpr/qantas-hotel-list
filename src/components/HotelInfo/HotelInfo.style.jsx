import styled from "styled-components";
import { SectionContainer } from "../../utils/styleUtils";
export const HotelInfoContainer = styled.div`
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: start;
    min-width: 79%;
`;

export const NameAndRatingContainer = styled.div`
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: start;
    align-items: baseline;
    padding: 0px;
    word-break: break-word;
    width: 100%;
`;
export const HotelNameAndAddressContainer = styled.div`
    display: flex;
    justify-content: start;
    flex-direction: column;
    word-break: break-word;
`;
export const StyledName = styled.p`
    white-space: nowrap;
    max-width: 445px;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 26px;
    font-weight: bold;
    margin-bottom: 0px;
`;
export const StyledAdress = styled.address`
    color: gray;
    text-decoration: none;
    word-break: break-word;
    font-size: 16px;
`;

export const StyledOffer = styled.p`
    color: red;
    text-decoration: underline;
    font-size: 16px;
`;
export const StyledCancellationPolicy = styled.p`
    color: green;
    font-size: 14px;
    margin: 0;
`;
