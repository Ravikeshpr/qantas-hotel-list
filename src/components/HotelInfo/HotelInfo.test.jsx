import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom/vitest";
import HotelInfo from "./HotelInfo";

describe("HotelInfo", () => {
    const props = {
        address: ["7-11 Talavera Rd", "North Ryde"],
        name: "Courtyard by Marriott Sydney-North Ryde",
        rating: {
            ratingValue: 4.5,
            ratingType: "self",
        },
        offerName: "Deluxe Balcony Room",
        cancellationPolicy: "NOT_REFUNDABLE",
    };
    it("HotelInfo renders as expected without props ", () => {
        const { container } = render(<HotelInfo />);
        expect(container.querySelector("address")).toBeEmptyDOMElement();
        expect(
            screen.queryByRole("7-11 Talavera Rd, North Ryde")
        ).not.toBeInTheDocument();
    });
    it("HotelInfo always renders same UI with correct props", () => {
        const { container } = render(<HotelInfo {...props} />);
        expect(container).toMatchSnapshot();
    });

    it("HotelInfo cancellationPolicy shows in Title case", () => {
        render(<HotelInfo {...props} />);
        expect(screen.getByText("Not Refundable")).toBeInTheDocument();
    });

    it("HotelInfo renders hotel Name with props", () => {
        render(<HotelInfo {...props} />);
        expect(
            screen.getByText("Courtyard by Marriott Sydney-North Ryde")
        ).toBeInTheDocument();
    });
    it("HotelInfo renders correct address with props", () => {
        render(<HotelInfo {...props} />);
        expect(
            screen.getByText("7-11 Talavera Rd, North Ryde")
        ).toBeInTheDocument();
    });

    it("HotelInfo renders correct offerName with props", () => {
        render(<HotelInfo {...props} />);
        expect(screen.getByText("Deluxe Balcony Room")).toBeInTheDocument();
    });
});
