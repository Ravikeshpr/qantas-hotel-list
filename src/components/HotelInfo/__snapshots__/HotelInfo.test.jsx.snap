// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HotelInfo > HotelInfo always renders same UI with correct props 1`] = `
<div>
  <div
    class="sc-gnOvAp cyQKIa"
  >
    <div
      class="sc-dQkurY knUWMK"
    >
      <div
        class="sc-glJfXD iPSiVB"
      >
        <p
          class="sc-dOfePm farNyc"
        >
          Courtyard by Marriott Sydney-North Ryde
        </p>
        <address
          class="sc-jgmqJh eBKsyD"
        >
          7-11 Talavera Rd, North Ryde
        </address>
      </div>
      <div
        class="sc-ihGilW gnCquM"
      >
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: yellow;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: yellow;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: yellow;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: yellow;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-half-stroke "
            data-icon="circle-half-stroke"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: yellow;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 256c0-106-86-192-192-192l0 384c106 0 192-86 192-192zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
    </div>
    <p
      class="sc-gIIpAQ gguqoy"
    >
      Deluxe Balcony Room
    </p>
    <p
      class="sc-kFkjun kFxDJK"
    >
      Not Refundable
    </p>
  </div>
</div>
`;
